test/e2e/cmd/workspaces_test.go

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/che-incubator/che-workspace-operator/test/e2e/pkg/config"
+	"github.com/che-incubator/che-workspace-operator/test/e2e/pkg/deploy"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/che-incubator/che-workspace-operator/test/e2e/pkg/client"
+	_ "github.com/che-incubator/che-workspace-operator/test/e2e/pkg/tests"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/reporters"
+	"github.com/onsi/gomega"
+)
+
+//Create Constant file
+const (
+	testResultsDirectory = "/tmp/artifacts"
+	jUnitOutputFilename  = "junit-workspaces-operator.xml"
+)
+
+//SynchronizedBeforeSuite blocks is executed before run all test suites
+var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
+	//!TODO: Try to create a specific function to call all <ginkgo suite> configuration.
+	fmt.Println("Starting to setup objects before run ginkgo suite")
+	config.Namespace = "che-workspace-controller"
+
+	workspaces, err := client.NewK8sClient()
+	if err != nil {
+		fmt.Println("Failed to create workspace client")
+	}
+
+	ns := newNamespace()
+	ns, err = workspaces.Kube().CoreV1().Namespaces().Create(ns)
+
+	if err != nil {
+		panic(err)
+	}
+
+	controller := deploy.NewDeployment(workspaces)
+
+	if err := controller.CreateAllOperatorRoles(); err != nil {
+		_ = fmt.Errorf("Failed to create roles in clusters %s", err)
+	}
+
+	if err := controller.CreateOperatorClusterRole(); err != nil {
+		_ = fmt.Errorf("Failed to create roles in clusters %s", err)
+	}
+
+	if err := controller.CustomResourceDefinitions(); err != nil {
+		_ = fmt.Errorf("Failed to add custom resources definitions to cluster %s", err)
```

Comment:
let's fail in any error, if deployment is failed - the tests are supposed to fail as well.